ingen/data_source/file_source.py

```diff
@@ -46,7 +46,7 @@ def fetch_data(self, reader):
         """
         returns a DataFrame of data fetched from input FileSource.
         """
-        return reader.read(self._src)
+        return reader.read(self._src, encoding=self._src['file_encoding'])
```

Comment:
this line returns following error
<img width="832" alt="Screenshot 2024-08-31 at 12 43 20 AM" src="https://github.com/user-attachments/assets/0dd49cd0-f487-4085-b725-39343d5d8382">

We don't need to pass encoding as a separate parameter, it can be included in `_src` itself